### PR TITLE
Add Push Report.

### PR DESCRIPTION
**Pull request by jpkrajewski**
# Pull Request Summary:
This pull request introduces a new feature called "Push Rapport" to the UX2Rollback model. The Push Rapport is designed to provide comprehensive information about the configuration push process to vManage. It includes details such as the success rate, created config groups, feature profiles, parcels, and any failed parcels with detailed debug messages.

# Description of Changes:
During the process of pushing configurations to vManage, the Push Rapport feature collects and stores valuable information about the various components created or modified. This includes:

- Config Groups: The Push Rapport keeps track of the configuration groups created during the push process.
- Feature Profiles: It also records the feature profiles that were created or updated.
- Parcels: The Push Rapport maintains a list of parcels that were successfully created, along with their respective UUIDs.
- Failed Parcels: In case of any failures during parcel creation, the Push Rapport captures detailed debug messages, providing valuable insights into the root cause of the issue.
- Success Rate: Additionally, the Push Rapport calculates and displays the overall success rate of the configuration push, represented as a percentage of successfully created parcels out of the total number of parcels.

The introduction of the Push Rapport feature enhances visibility and transparency during the configuration push process. 

# Example
image
image
image

# Checklist:
- [X] Make sure to run pre-commit before committing changes
- [X] Make sure all checks have passed
- [X] PR description is clear and comprehensive
- [X] Mentioned the issue that this PR solves (if applicable)
- [X] Make sure you test the changes

### CODE REVIEW

**Review by smialy — APPROVED** [review]

File: catalystwan/models/configuration/config_migration.py [hunk 1]
```diff
@@ -167,6 +188,89 @@ def insert_parcel_type_from_headers(cls, values: Dict[str, Any]):
...
+    def add_rapport(self, name: str, uuid: UUID, feature_profiles: List[FeatureProfileBuildRaport]) -> None:
+        self.config_groups.append(ConfigGroupRaport(name=name, uuid=uuid, feature_profiles=feature_profiles))
+
+    def set_push_success_rate(self):
+        for config_group in self.config_groups:
+            for feature_profile in config_group.feature_profiles:
+                self.created_parcels += len(feature_profile.created_parcels)
+                self.failed_parcels += len(feature_profile.failed_parcels)
+        all_parcels = self.created_parcels + self.failed_parcels
+        self.success_rate_message = (
```
Comment (radkrawczyk) [inline catalystwan/models/configuration/config_migration.py:260]: the member 'success_rate_message' is not used anywhere, is that right?
Also wondering, if wouldn't be better to not store this message in class member, but just return the current stats by 'get_success_rate_message' or sth similar

**Review by radkrawczyk — COMMENTED** [review]
The 'repport' string occurs many times ~20, probably should be renamed to 'raport' ;)

File: catalystwan/api/builders/feature_profiles/handler.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import functools
+from typing import Optional
+from uuid import UUID
+
+from catalystwan.exceptions import ManagerHTTPError
+from catalystwan.models.configuration.feature_profile.parcel import AnyParcel
+
+
+def handle_build_rapport(func):
```
[the remark this answers is not in the document]
Comment (jpkrajewski) [reply catalystwan/api/builders/feature_profiles/handler.py]: Changed

File: catalystwan/api/configuration_groups/parcel.py [hunk 3]
```diff
@@ -17,6 +17,8 @@
 from catalystwan.exceptions import CatalystwanException
 from catalystwan.models.common import VersionedField
 
+# from catalystwan.models.configuration.feature_profile.parcel import ParcelType
```
[the remark this answers is not in the document]
Comment (jpkrajewski) [reply catalystwan/api/configuration_groups/parcel.py]: Removed

**Review by radkrawczyk — APPROVED** [review]

File: catalystwan/models/configuration/config_migration.py (hunk 1 above)
Comment (jpkrajewski) [reply catalystwan/models/configuration/config_migration.py:260]: This field will be send to the frontend

File: catalystwan/api/builders/feature_profiles/handler.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (sbasan) [reply catalystwan/api/builders/feature_profiles/handler.py]: I would say it should be report with 'e' :)

**Review by sbasan — APPROVED** [review]

**Review by sbasan — APPROVED** [review]

**Pull request merged**